zarr/util.py

```diff
@@ -170,29 +166,11 @@ def normalize_array_selection(item, shape):
     """Convenience function to normalize a selection within an array with
     the given `shape`."""
 
-    # normalize item
-    if isinstance(item, integer_types):
-        item = (int(item),)
-    elif isinstance(item, slice):
-        item = (item,)
-    elif item == Ellipsis:
-        item = (slice(None),)
-
-    # handle tuple of indices/slices
-    if isinstance(item, tuple):
+    rf_item = reformat_slices(item)
+    if Ellipsis not in rf_item:
+        rf_item += (Ellipsis,)
```

Comment:
This shouldn't be necessary, but is really a bug on my part. Will fix in `kenjutsu` with PR ( https://github.com/jakirkham/kenjutsu/pull/59 ). In the interim, this shouldn't cause any issues and should be fine even when we fix it.